spec/shared_contexts/checksum.rb

```diff
@@ -0,0 +1,48 @@
+# Shared contexts for testing against all supported checksum types.
+#
+# These helpers define nested rspec example groups to test code against all our
+# supported checksum types. Example groups that need to be run against all
+# types should use the `with_checksum_types` helper which will
+# create a new example group for each types and will run the given block
+# in each example group.
+
+CHECKSUM_STAT_TIME = Time.now
+CHECKSUM_TYPES_TO_TRY = [
+  ['md5', 'a7a169ac84bb863b30484d0aa03139c1'],
+  ['md5lite', '22b4182363e81b326e98231fde616782'],
+  ['sha256', '47fcae62967db2fb5cba2fc0d9cf3e6767035d763d825ecda535a7b1928b9746'],
+  ['sha256lite', 'fd50217a2b0286ba25121bf2297bbe6c197933992de67e4e568f19861444ecf8'],
+  ['ctime', "#{CHECKSUM_STAT_TIME}"],
+  ['mtime', "#{CHECKSUM_STAT_TIME}"]
+]
+
+shared_context('with supported checksum types') do
```

Comment:
I'm not a huge fan of shared contexts, because data (really memoized methods) are magically available inside of your test, like [`checksum`](https://github.com/puppetlabs/puppet/pull/3507/files#diff-ff71306e757bf24c8e36adb4bc7da8cfR33) below. It gets really confusing if your test also has a locally defined `let(:checksum)`.

We've been moving more towards things like `/spec/lib/puppet_spec/compiler.rb`. It's more explicit and less magical. That way you can include the module in your test, call the helper methods, passing all of the required data.